### ISSUE TITLE
BigDecimal scale and precision when suppling a schema

### ISSUE THREAD

**Issue by sebastianvoss** [sksamuel/avro4s#267]
I'm using the Confuent schema registry to store my schema and running into issues when reading records with `BigDecimal` values. I would like to use the schema supplied by the registry rather than letting `avro4s` generate it. But when I supply a schema to `RecordFormat[Measurement](schema)` it seems it is not repecting `scale` and `precision` for logical type `decimal`.

Here is an example:

```
case class Measurement(value: BigDecimal)

val deserializer = new GenericAvroDeserializer()
val record = deserializer.deserialize("test", bytes)
val schema = record.getSchema
val format = RecordFormat[Measurement](schema)
val m = format.from(record)
```
When I'm running the same code with `implicit val sp = ScalePrecisionRoundingMode(8, 20, BigDecimal.RoundingMode.HALF_UP)` in scope it gives me different results (e.g. `123456789.12312313` with `sp` in scope  and `123456789123123.13` without).

In the registry, the schema for field `value` is:
```
{
    "name" : "value",
    "type" : {
      "type" : "bytes",
      "logicalType" : "decimal",
      "precision" : 20,
      "scale" : 8
    }
}
```

Shouldn't `avro4s` take the precision and scale from the supplied schema and not requiring an implicit `ScalePrecisionRoundingMode` in this case?

**Comment by sksamuel** [comment]
Yes that's right. The Decoder should use the values in the schema.
Only the schema generator should use `ScalePrecisionRoundingMode`.

**Comment by sksamuel** [comment]
Fixed in 2.0.4
